variants/aws-dev/Cargo.toml

```diff
@@ -28,3 +28,21 @@ included-packages = [
 
```

Comment:
I am sad about the duplication here. Could we find the list of direct dependencies somehow instead?

Reply:
You mean the duplication between our own metadata in `included-packages` and Cargo's dependency list? Unfortunately it's not exact duplication. In `included-packages` we're telling buildsys what to install, and in Cargo's list we're telling Cargo what has to be built in order for those packages to be available. It's not always 1:1.

There are some designs we could consider to cut down on the duplication.
- Introduce a package/spec for each variant, e.g. `aws-dev.spec` where Cargo.toml would give the dependencies for Cargo and the spec would say `Requires` for each of the dependencies. (?)
- Some kind of preprocessing of the variant's Cargo.toml to figure out what to build during the variant's `build.rs`? I'm not sure how this would work, and I'm not sure how buildsys would understand when a subpackage comes from a parent package.

Any ideas?

Reply:
It looks like the exceptions all come from subpackages of `os.spec`, so one alternative would be for the `os` package to conditionally include its own subpackages based on the variant.

I didn't like the idea at first, but we're already conditionalizing the building of those components and their subpackages within `os.spec`, so it isn't making things any more weird than they already are. It would also thin out dependencies in `release.spec` quite a bit - it'd just depend on `os` - and remove some of the coupling between those two.

We'd end up with an implicit rule like "all main packages must require all of their own subpackages that they need at runtime."